app/labor/markdown_parser.rb

```diff
@@ -32,8 +32,8 @@ def finalize(link_attributes: {})
     rescue Liquid::SyntaxError => e
       html = e.message
     end
-    html = remove_nested_linebreak_in_list(html)
-    html = prefix_all_images(html)
+    html = Html::RemoveNestedLinebreakInList.call(html)
```

Comment:
Are we calling this anywhere else? Right now it seems only `Html::PrefixAllImages` has a separate caller (`app/models/display_ad.rb`), so maybe we could inline this similar to e.g. `wrap_all_images_in_links`? 

Of course, this also depends on what your goals for refactoring the Markdown parser itself are. I could see a use case for for a `HtmlProcessor` object with a fluent interface (for chaining), something like

```ruby
class HtmlProcessor
  attr_reader :html

  def initialize(html)
    @html = html
  end

  def remove_nested_linebreak_in_list
    ...
    self
  end
  
   ....
end
```

This would turn the assignment cascade here into something like

```ruby
html = HtmlProcessor.
  new(html).
  remove_nested_linebreak_in_list.
  prefix_all_images.
  html
```

Anyway, just a random idea, pick what you think is best 😃